src/components/ServiceQuestions.js

```diff
@@ -0,0 +1,245 @@
+        var pageText = eID.options[eID.selectedIndex].text;
+
+        if (pageText === "10") {
+            this.setState({itemsPerPage: 10, pageNumber: 1})
+        }
+        if (pageText === "25") {
+            this.setState({itemsPerPage: 25, pageNumber: 1})
+        }
+        if (pageText === "50") {
+            this.setState({itemsPerPage: 50, pageNumber: 1})
+        }
+        if (pageText === "All") {
+            this.setState({itemsPerPage: 1000000, pageNumber: 1})
+        }
+    }
+
+    previousPage() {
+        if (this.state.pageNumber !== 1) {
+            var temp = this.state.pageNumber - 1;
+            this.setState({pageNumber: temp })
+        }
+    }
+
+    nextPage() {
+        var totalPages = Math.ceil(this.state.serviceQuestions.length / this.state.itemsPerPage);
+        if (this.state.pageNumber + 1 <= totalPages) {
+            var temp = this.state.pageNumber + 1;
+            this.setState({pageNumber: temp })
+        }
+    }
+
+    choosePage(newPage) {
+        this.setState({pageNumber: newPage })
+
+    }
+
+    createPageButtons() {
+        var totalPages = Math.ceil(this.state.serviceQuestions.length / this.state.itemsPerPage);
+        var i;
+
+        for (i = 1; i < totalPages + 1; i++) {
+            if ((document.getElementById("Button " + i)) == null)
+            {
+                console.log(i);
+                var button = document.createElement("BUTTON");
+                var buttonText = document.createTextNode("" + i);
+                button.appendChild(buttonText);
+                button.type = "button";
+                button.id = "Button " + i;
+                button.value = i;
+                button.onclick = function(e) {
+                    this.setState({pageNumber: e.target.value})
+                }.bind(this);
+                document.getElementById("pageButtons").appendChild(button);
+            }
+        }
+    }
+
+    addServiceQuestion() {
+        console.log('Add Service Click happened');
+        const titleInput = document.getElementById('titleInput');
+        const questionInput = document.getElementById('questionInput');
+        var title = titleInput.value;
+        var question = questionInput.value;
+        if (title === "" || question === "") {
+            alert("Please add a question")
+        }
+        this.serviceQuestionsService
+            .addServiceQuestion({title: title, question: question})
+            .then(() => this.refreshServiceQuestions());
+        titleInput.value = "";
+        questionInput.value = "";
+    }
+
+    updateServiceQuestion(id) {
+        const titleInput = document.getElementById('titleInput');
+        const questionInput = document.getElementById('questionInput');
+        var title = titleInput.value;
+        var question = questionInput.value;
+        this.serviceQuestionsService
+            .updateServiceQuestion({title: title, question: question}, id)

```

Comment:
Same comment as below for the update. These aren't the columns for the ServiceQuestion. I'm also getting a cross-origin error for some reason so I can't test it. I don't know if you're seeing the same or not thought it might just be me.